File: backend/app/api/admin/notifications.py
"""
app/api/admin/notifications.py

Router for admin notification management.
Handles listing all notifications and marking individual
or multiple notifications as read, as well as bulk deletion.
"""

from fastapi import APIRouter, Depends, Body
from sqlalchemy.orm import Session
from app.api.deps import get_db
from app.schemas.notification import NotificationOut
from app.services.notification_service import (
    get_all_notifications,
    mark_read,
    mark_many_read,
    delete_many,
)

router = APIRouter(prefix="/notifications")


# =================================================================================
# NOTIFICATIONS
# =================================================================================

@router.get("", response_model=list[NotificationOut])
def list_notifications(db: Session = Depends(get_db)):
    return get_all_notifications(db)


@router.patch("/{notification_id}/read")
def mark_notification_read(notification_id: int, db: Session = Depends(get_db)):
    mark_read(db, notification_id)
    return {"detail": "Marked as read"}


# =================================================================================
# BULK OPERATIONS
# =================================================================================

@router.post("/mark-read")
def bulk_mark_read(ids: list[int] = Body(...), db: Session = Depends(get_db)):
    count = mark_many_read(db, ids)
    return {"detail": f"{count} notifications marked as read"}


@router.post("/bulk-delete")
def bulk_delete_notifications(ids: list[int] = Body(...), db: Session = Depends(get_db)):
    count = delete_many(db, ids)
    return {"detail": f"{count} notifications deleted"}

File: admin-dashboard/src/api/notifications.js
import api from './http'

export const getAllNotifications = async () => {
  try {
    const response = await api.get('/admin/notifications')
    return response.data
  } catch (err) {
    console.error('Failed to fetch notifications:', err)
    throw err
  }
}

export const markNotificationRead = async (notificationId) => {
  try {
    const response = await api.patch(`/admin/notifications/${notificationId}/read`)
    return response.data
  } catch (err) {
    console.error('Failed to mark notification as read:', err)
    throw err
  }
}

export const bulkMarkRead = async (ids) => {
  try {
    const response = await api.post('/admin/notifications/mark-read', ids)
    return response.data
  } catch (err) {
    console.error('Failed to bulk mark notifications as read:', err)
    throw err
  }
}

export const bulkDeleteNotifications = async (ids) => {
  try {
    const response = await api.post('/admin/notifications/bulk-delete', ids)
    return response.data
  } catch (err) {
    console.error('Failed to bulk delete notifications:', err)
    throw err
  }
}